libraries/ESP8266WebServer/src/ESP8266WebServer.cpp

```diff
@@ -127,74 +137,74 @@ bool ESP8266WebServer::authenticate(const char * username, const char * password
       }
       delete[] toencode;
       delete[] encoded;
-    }else if(authReq.startsWith("Digest")){
+    } else if(authReq.startsWith(F("Digest"))) {
       authReq = authReq.substring(7);
       #ifdef DEBUG_ESP_HTTP_SERVER
       DEBUG_OUTPUT.println(authReq);
       #endif
-      String _username = _exractParam(authReq,"username=\"");
-      if((!_username.length())||_username!=String(username)){
+      String _username = _extractParam(authReq,F("username=\""));
+      if(!_username.length() || _username != String(username)) {
         authReq = String();
         return false;
       }
       // extracting required parameters for RFC 2069 simpler Digest
-      String _realm    = _exractParam(authReq,"realm=\"");
-      String _nonce    = _exractParam(authReq,"nonce=\"");
-      String _uri      = _exractParam(authReq,"uri=\"");
-      String _response = _exractParam(authReq,"response=\"");
-      String _opaque   = _exractParam(authReq,"opaque=\"");
+      String _realm    = _extractParam(authReq, F("realm=\""));
+      String _nonce    = _extractParam(authReq, F("nonce=\""));
+      String _uri      = _extractParam(authReq, F("uri=\""));
+      String _response = _extractParam(authReq, F("response=\""));
+      String _opaque   = _extractParam(authReq, F("opaque=\""));
 
-      if((!_realm.length())||(!_nonce.length())||(!_uri.length())||(!_response.length())||(!_opaque.length())){
+      if((!_realm.length()) || (!_nonce.length()) || (!_uri.length()) || (!_response.length()) || (!_opaque.length())) {
         authReq = String();
         return false;
       }
-      if((_opaque!=_sopaque)||(_nonce!=_snonce)||(_realm!=_srealm)){
+      if((_opaque != _sopaque) || (_nonce != _snonce) || (_realm != _srealm)) {
         authReq = String();
         return false;
       }
       // parameters for the RFC 2617 newer Digest
       String _nc,_cnonce;
-      if(authReq.indexOf("qop=auth") != -1){
-        _nc = _exractParam(authReq,"nc=",',');
-        _cnonce = _exractParam(authReq,"cnonce=\"");
+      if(authReq.indexOf(FPSTR(qop_auth)) != -1) {
+        _nc = _extractParam(authReq, F("nc="), ',');
```

Comment:
Thanks for fixing this typo in the method name!  _exractParam always gave me a headache!